ui/app/src/components/ContentTypeManagement/components/CreateTypeDialog.tsx

```diff
@@ -58,33 +67,74 @@ const prefixes = {
 	component: '/component/'
 };
 
-function CreateTypeDialogBody({ onAccept }: CreateTypeDialogBaseProps) {
+function CreateTypeDialogBody(props: CreateTypeDialogBaseProps) {
+	const { onAccept, onSubmittingAndOrPendingChange } = props;
+	const siteId = useActiveSiteId();
 	const [type, setType] = useState<'page' | 'component'>('page');
 	const [name, setName] = useState<string>('');
 	const [id, setId] = useState<string>('');
-	const [prefix, setPrefix] = useState<string>('');
+	const prefix = useRef<string>(undefined);
+	prefix.current = prefixes[type];
 	const dialogContext = useEnhancedDialogContext();
+	const contentTypes = useContentTypes();
+	const [nameExists, setNameExists] = useState<boolean>(false);
+	const [idExists, setIdExists] = useState<boolean>(false);
+	const isValid = useMemo(() => {
+		return validate({ id, name, type, contentTypes, setNameExists, setIdExists });
+	}, [id, name, type, contentTypes]);
+	const enableSubmit = Boolean(isValid) && Boolean(name) && Boolean(id) && Boolean(type);
+	const [fetchingContentTypes, setFetchingContentTypes] = useState(false);
+	const dispatch = useDispatch();
+	const [idManuallyChanged, setIdManuallyChanged] = useState(false);
+
 	const validateAndSubmit = () => {
-		// TODO: Validate before accepting
-		if (!validate(id, name, type)) return;
```

Comment:
Why remove the validation?

Reply:
Validation was moved down since it now also checks against list of content types